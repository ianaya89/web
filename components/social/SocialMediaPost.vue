<template>
  <div class="social-media-post" @click="$refs.socialPostModal.open()">
    <div class="w-full py-3 px-2 flex">
      <avatar :src="$get(post,'socialAccount.profilePicture.url')" :classes="'w-10 h-10'" />
      <div class="ml-2 my-auto">
        <span class="text-h7 font-medium">
          {{ $get(post,'socialAccount.name', '-') }}
        </span>
        <!--        <p class="text-grey">-->
        <!--          @tag-->
        <!--        </p>-->
      </div>
    </div>
    <img
      v-if="hasMedia"
      class="w-full"
      :src="$get(post,'thumb.url', undefined)"
      alt="Social media post cover image"
    >
    <div class="w-full px-3">
      <span class="font-serif">
        {{ $get(post, 'content','-') | truncate(hasMedia ? 80 : 200) }}
      </span>
      <social-media-post-footer v-if="socialType!=='web-page'" :post="post" :social-type="socialType" />
    </div>
    <social-media-post-modal ref="socialPostModal" :social-post="post" :social-type="socialType" />
  </div>
</template>

<script>
import Avatar from '@/components/Avatar'
import SocialMediaPostFooter from '@/components/social/SocialMediaPostFooter'
import SocialMediaPostModal from '@/components/modals/SocialMediaPostModal'

export default {
  name: 'SocialMediaPost',

  components: {
    SocialMediaPostModal,
    SocialMediaPostFooter,
    Avatar
  },

  props: {
    post: {
      type: Object,
      required: true
    },
    socialType: {
      type: String,
      required: true
    }
  },

  computed: {
    hasMedia() {
      return (
        this.$get(this.post, 'type', null) === 'image' ||
        this.$get(this.post, 'type', null) === 'video'
      )
    }
  }
}
</script>

<style lang="postcss" scoped>
.social-media-post {
  @apply .bg-white .border .border-grey-l4 .cursor-pointer;
  transition: box-shadow 300ms ease-in;
  will-change: box-shadow;

  &:hover {
    @apply .shadow-lg;
  }
}
</style>
