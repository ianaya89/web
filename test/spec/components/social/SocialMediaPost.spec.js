import { shallowMount } from '@vue/test-utils'
import SocialMediaPost from '@/components/social/SocialMediaPost'

describe('SocialMediaPost', () => {
  it('renders SocialMediaPost', () => {
    const wrapper = shallowMount(SocialMediaPost, {
      propsData: {
        post: {
          type: 'video'
        },
        socialType: 'facebook'
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('checks if post has media', () => {
    const wrapper = shallowMount(SocialMediaPost, {
      propsData: {
        post: {
          type: 'video'
        },
        socialType: 'facebook'
      }
    })

    expect(wrapper.vm.hasMedia).toBeTruthy()

    wrapper.setProps({
      post: {
        type: 'image'
      }
    })

    expect(wrapper.vm.hasMedia).toBeTruthy()

    wrapper.setProps({
      post: {
        type: 'text'
      }
    })

    expect(wrapper.vm.hasMedia).toBeFalsy()
  })
})
