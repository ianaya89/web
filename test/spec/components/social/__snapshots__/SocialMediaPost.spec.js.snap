// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaPost renders SocialMediaPost 1`] = `
<div
  class="social-media-post"
>
  <div
    class="w-full py-3 px-2 flex"
  >
    <avatar-stub
      avatarclasses=""
      classes="w-10 h-10"
      rounded="true"
      src=""
      tooltipcolor="light"
      tooltipposition="top"
    />
     
    <div
      class="ml-2 my-auto"
    >
      <span
        class="text-h7 font-medium"
      >
        
        -
      
      </span>
    </div>
  </div>
   
  <img
    alt="Social media post cover image"
    class="w-full"
    src=""
  />
   
  <div
    class="w-full px-3"
  >
    <span
      class="font-serif"
    >
      
      -
    
    </span>
     
    <social-media-post-footer-stub
      post="[object Object]"
      socialtype="facebook"
    />
  </div>
   
  <social-media-post-modal-stub
    socialpost="[object Object]"
    socialtype="facebook"
  />
</div>
`;
